instant-gram/ViewControllers/LoginViewController.m

```diff
@@ -0,0 +1,106 @@
+//
+//  LoginViewController.m
+//  instant-gram
+//
+//  Created by Joey R. Hernandez Perez on 7/7/20.
+//  Copyright © 2020 joeyrhernandez1. All rights reserved.
+//
+
+#import "LoginViewController.h"
+#import <Parse/Parse.h>
+
+@interface LoginViewController ()
+
+@property (weak, nonatomic) IBOutlet UITextField *usernameField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordField;
+
+@end
+
+@implementation LoginViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+}
+
+- (IBAction)OnTapRegister:(id)sender {
+    [self registerUser];
+}
+
+- (void)registerUser {
+    PFUser *newUser = [PFUser user];
+    
+    if ([self.usernameField.text isEqual:@""] || [self.passwordField.text isEqual:@""] ) {
+        [self showAlert:@""];
+        return;
+    }
+    
+    newUser.username = self.usernameField.text;
+    newUser.password = self.passwordField.text;
+    
+    [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * error) {
+        if (error != nil) {
+            NSLog(@"Error: %@", error.localizedDescription);
+            [self showAlert:error.localizedDescription];
+        } else {
+            NSLog(@"User registered successfully");
+        }
+    }];
+}
+
+- (IBAction)onTapSignIn:(id)sender {
+    [self loginUser];
+}
+
+- (void)loginUser {
+    if ([self.usernameField.text isEqual:@""] || [self.passwordField.text isEqual:@""] ) {
```

Comment:
This repetition could be abstracted to a computer property like, "areUserInputFieldsEmpty" or something.